dhctl/pkg/server/pkg/helper/ssh_client.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2024 Flant JSC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helper
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/deckhouse/deckhouse/dhctl/pkg/app"
+	"github.com/deckhouse/deckhouse/dhctl/pkg/config"
+	"github.com/deckhouse/deckhouse/dhctl/pkg/operations/bootstrap"
+	"github.com/deckhouse/deckhouse/dhctl/pkg/server/pkg/util"
+	"github.com/deckhouse/deckhouse/dhctl/pkg/server/pkg/util/callback"
+	"github.com/deckhouse/deckhouse/dhctl/pkg/system/node/ssh"
+	"github.com/deckhouse/deckhouse/dhctl/pkg/system/node/ssh/session"
+)
+
+func CreateSSHClient(config *config.ConnectionConfig) (*ssh.Client, func() error, error) {
+	cleanuper := callback.NewCallback()
+
+	keysPaths := make([]string, 0, len(config.SSHConfig.SSHAgentPrivateKeys))
+	for _, key := range config.SSHConfig.SSHAgentPrivateKeys {
+		keyPath, cleanup, err := util.WriteDefaultTempFile([]byte(strings.TrimSpace(key.Key) + "\n"))
```

Comment:
i think copying private keys in temp directory (or another directory) it is not security.